extensions/pkg/controller/healthcheck/general/managed_resource.go

```diff
@@ -79,10 +79,17 @@ func (healthChecker *ManagedResourceHealthChecker) Check(ctx context.Context, re
 	}
 	if isHealthy, err := managedResourceIsHealthy(mcmDeployment); !isHealthy {
 		healthChecker.logger.Error(err, "Health check failed")
+
+		var errorCode []gardencorev1beta1.ErrorCode
+		configurationProblemRegexp := regexp.MustCompile(`(?i)(notFound|unrecognized feature gate|strict decoder error|not allowed to configure an unsupported|error during apply of object .* is invalid:)`)
```

Comment:
Can you explain why you included

- notFound
- unrecognized feature gate
- not allowed to configure an unsupported

here?